File: dart/lib/questionnaire/questions.dart
import "package:dart/questionnaire/choice.dart";
import "package:dart/questionnaire/question.dart";

var frontendQuestions = [
  Question(
      "What comes to your mind when I say 'Svelte'?",
      [
        Choice("A", "Just another Javascript framework"),
        Choice("B", "It’s a new language"),
        Choice("C", "Smallest bundle size"),
        Choice("D", "Not sure man"),
      ],
      "C",
      "frontend"),
  Question(
      "What's used to style HTML files?",
      [
        Choice("A", "Condensed Style Sheets"),
        Choice("B", "Cascaded Style Scripts"),
        Choice("C", "Cascaded Style Sheet"),
        Choice("D", "I don't know my guy"),
      ],
      "C",
      "frontend"),
  Question(
      "Google applications mostly use Material design for their UI",
      [
        Choice("A", "Yes, they do"),
        Choice("B", "Nah, they use Bootstrap"),
        Choice("C", "They use Tailwind CSS"),
        Choice("D", "I have no idea about this"),
      ],
      "A",
      "frontend"),
  Question(
      "The Angular framework was created by Google",
      [
        Choice("A", "Definitely"),
        Choice("B", "No, it’s by Facebook"),
        Choice("C", "It’s developed by Netflix for their UIs"),
        Choice("D", "I don’t know much about frontend development frameworks"),
      ],
      "A",
      "frontend"),
  Question(
      "What does React, Vue and Angular have in common?",
      [
        Choice("A", "They are languages"),
        Choice("B", "They're frameworks from different languages"),
        Choice("C", "All compile to HTML, CSS and JS files"),
        Choice("D", "I have no idea about these guys"),
      ],
      "C",
      "frontend")
];

var backendQuestions = [
  Question(
      "Ever heard of 'Laravel’? What's it used for?",
      [
        Choice("A", "Development of the backend applications"),
        Choice("B", "Native applications"),
        Choice("C", "Frontend development"),
        Choice("D", "Never heard of it"),
      ],
      "A",
      "backend"),
  Question(
      "Redis cannot be used as a standalone server database",
      [
        Choice("A", "It can"),
        Choice("B", "Redis is not a database"),
        Choice("C", "True"),
        Choice("D", "I don't know what Redis is"),
      ],
      "C",
      "backend"),
  Question(
      "How would you use caching to reduce your server response times?",
      [
        Choice("A", "Cache common requests"),
        Choice("B", "It's not a way to reduce response times"),
        Choice("C", "Cache all responses"),
        Choice("D", "I don't know the answer to this"),
      ],
      "A",
      "backend"),
  Question(
      "Which of the statements below is true when it comes to Docker?",
      [
        Choice("A", "It's an alternative to Kubernetes"),
        Choice("B", "Used for containerization of applications"),
        Choice("C", "Used to monitor backend applications"),
        Choice("D", "I have no idea what Docker is"),
      ],
      "B",
      "backend"),
  Question(
      "What's the largest file size that can be carried in a single request?",
      [
        Choice("A", "50MB"),
        Choice("B", "10MB"),
        Choice("C", "100MB"),
        Choice("D", "Not sure man"),
      ],
      "A",
      "backend")
];

var devopsQuestions = [
  Question(
      "Travis CI and Github actions serve almost the same functionality",
      [
        Choice("A", "I don't think so"),
        Choice("B", "Yessir, they do"),
        Choice("C", "Github actions is a version control system"),
        Choice("D", "I don't know these two technologies"),
      ],
      "B",
      "devops"),
  Question(
      "Docker was here long before Kubernetes",
      [
        Choice("A", "This is a big ass lie"),
        Choice("B", "That's a fact"),
        Choice("C", "Both were developed around the same time"),
        Choice("D", "I have no idea what Docker or Kubernetes is"),
      ],
      "A",
      "devops"),
  Question(
      "Prometheus is famous for monitoring server applications",
      [
        Choice("A", "I agree"),
        Choice("B", "Statement is false"),
        Choice("C", "It only collects information"),
        Choice("D", "Didn't catch that"),
      ],
      "A",
      "devops"),
  Question(
      "Kali is a Linux distro",
      [
        Choice("A", "Not!"),
        Choice("B", "It's a flavor of Ubuntu"),
        Choice("C", "It is!"),
        Choice("D", "I don't mess with Linux"),
      ],
      "C",
      "devops"),
  Question(
      "What does Google Cloud Platform, Azure and Heroku have in common?",
      [
        Choice("A", "All have IaaS solutions"),
        Choice("B", "They're the top cloud providers"),
        Choice("C", "They all provide PaaS solutions"),
        Choice("D", "They nothing in common"),
      ],
      "C",
      "devops")
];

var mobileQuestions = [
  Question(
      "It's cheaper to host my applications on Apple app store than Google play store",
      [
        Choice("A", "That’s a lie my guy"),
        Choice("B", "Sure, Apple offer the cheapest hosting for mobile apps"),
        Choice("C", "They offer almost the same pricing"),
        Choice("D", "I don't host my apps here"),
      ],
      "A",
      "mobile"),
  Question(
      "Which one of the following describes what Swift does?",
      [
        Choice("A", "It’s used for developing iOS apps"),
        Choice("B", "It can be used for developing Safari web extensions"),
        Choice("C", "Second name of a very famous music artist"),
        Choice("D", "I don’t know anything about Swift"),
      ],
      "A",
      "mobile"),
  Question(
      "Cross platform development allows me to implement one design for all my apps",
      [
        Choice("A",
            "You have to implement different designs using native UI SDKs"),
        Choice("B", "This is true, the design will be the same for all apps"),
        Choice("C", "It’s one codebase but designs come out differently"),
        Choice("D", "I have no idea"),
      ],
      "B",
      "mobile"),
  Question(
      "Large scale mobile applications can be built without user journeys and design prototypes",
      [
        Choice("A", "User journeys are not required for large projects"),
        Choice(
            "B", "That’s impossible, user’s journeys should be a prerequisite"),
        Choice("C",
            "Design prototypes are created at the same time an MVP is developed"),
        Choice("D", "I don’t do mobile development"),
      ],
      "B",
      "mobile"),
  Question(
      "Which one of the following best describes what Java SDK is for?",
      [
        Choice("A", "It’s for developing backend applications"),
        Choice("B", "It’s for developing different versions of Java"),
        Choice("C", "It’s used for android app development"),
        Choice("D", "I’m not well versed with Java"),
      ],
      "C",
      "mobile")
];

File: lib/r/src/questionnaire/questions.r
source("questionnaire/question.r")
source("questionnaire/choice.r")

frontendQuestions<- c(
    QuestionnaireQuestion(
        statement="What comes to your mind when I say 'Svelte'?",
        choices= c(
            QuestionnaireChoice(letter="A", statement="Just another Javascript framework"),
            QuestionnaireChoice(letter="B", statement="It's a new language"),
            QuestionnaireChoice(letter="C", statement="Smallest bundle size"),
            QuestionnaireChoice(letter="D", statement="Not sure man")
        ),
        answer="C",
        tag="frontend"
    ),
    QuestionnaireQuestion(
        statement="What's used to style HTML files?",
        choices= c(
            QuestionnaireChoice(letter="A", statement="Condensed Style Sheets"),
            QuestionnaireChoice(letter="B", statement="Cascaded Style Scripts"),
            QuestionnaireChoice(letter="C", statement="Cascaded Style Sheet"),
            QuestionnaireChoice(letter="D", statement="I don't know my guy")
        ),
        answer="C",
        tag="frontend"
    ),
    QuestionnaireQuestion(
        statement="Google applications mostly use Material design for their UI",
        choices= c(
            QuestionnaireChoice(letter="A", statement="Yes, they do"),
            QuestionnaireChoice(letter="B", statement="Nah, they use Bootstrap"),
            QuestionnaireChoice(letter="C", statement="They use Tailwind CSS"),
            QuestionnaireChoice(letter="D", statement="I have no idea about this")
        ),
        answer="A",
        tag="frontend"
    ),
    QuestionnaireQuestion(
        statement="The Angular framework was created by Google",
        choices= c(
            QuestionnaireChoice(letter="A", statement="Definitely"),
            QuestionnaireChoice(letter="B", statement="No, it's by Facebook"),
            QuestionnaireChoice(letter="C", statement="It's developed by Netflix for their UIs"),
            QuestionnaireChoice(
                letter="D", statement="I don't know much about frontend development frameworks")
        ),
        answer="A",
        tag="frontend"
    ),
    QuestionnaireQuestion(
        statement="What does React, Vue and Angular have in common?",
        choices= c(
            QuestionnaireChoice(letter="A", statement="They are languages"),
            QuestionnaireChoice(letter="B", statement="They're frameworks from different languages"),
            QuestionnaireChoice(letter="C", statement="All compile to HTML, CSS and JS files"),
            QuestionnaireChoice(letter="D", statement="I have no idea about these guys")
        ),
        answer="C",
        tag="frontend"
    )
)


backendQuestions<- c(
    QuestionnaireQuestion(
        statement="Ever heard of 'Laravel'? What's it used for?",
        choices= c(
            QuestionnaireChoice(letter="A", statement="Development of the backend applications"),
            QuestionnaireChoice(letter="B", statement="Native applications"),
            QuestionnaireChoice(letter="C", statement="Frontend development"),
            QuestionnaireChoice(letter="D", statement="Never heard of it")
        ),
        answer="A",
        tag="backend"
    ),
    QuestionnaireQuestion(
        statement="Redis cannot be used as a standalone server database",
        choices= c(
            QuestionnaireChoice(letter="A", statement="It can"),
            QuestionnaireChoice(letter="B", statement="Redis is not a database"),
            QuestionnaireChoice(letter="C", statement="True"),
            QuestionnaireChoice(letter="D", statement="I don't know what Redis is")
        ),
        answer="C",
        tag="backend"
    ),
    QuestionnaireQuestion(
        statement="How would you use caching to reduce your server response times?",
        choices= c(
            QuestionnaireChoice(letter="A", statement="Cache common requests"),
            QuestionnaireChoice(letter="B", statement="It's not a way to reduce response times"),
            QuestionnaireChoice(letter="C", statement="Cache all responses"),
            QuestionnaireChoice(letter="D", statement="I don't know the answer to this")
        ),
        answer="A",
        tag="backend"
    ),
    QuestionnaireQuestion(
        statement="Which of the statements below is true when it comes to Docker?",
        choices= c(
            QuestionnaireChoice(letter="A", statement="It's an alternative to Kubernetes"),
            QuestionnaireChoice(letter="B", statement="Used for containerization of applications"),
            QuestionnaireChoice(letter="C", statement="Used to monitor backend applications"),
            QuestionnaireChoice(
                letter="D", statement="I have no idea what Docker is")
        ),
        answer="B",
        tag="backend"
    ),
    QuestionnaireQuestion(
        statement="What's the largest file size that can be carried in a single request?",
        choices= c(
            QuestionnaireChoice(letter="A", statement="50MB"),
            QuestionnaireChoice(letter="B", statement="10MB"),
            QuestionnaireChoice(letter="C", statement="100MB"),
            QuestionnaireChoice(letter="D", statement="Not sure man")
        ),
        answer="A",
        tag="backend"
    )
)


devopsQuestions<- c(
    QuestionnaireQuestion(
        statement="Travis CI and Github actions serve almost the same functionality",
        choices= c(
            QuestionnaireChoice(letter="A", statement="I don't think so"),
            QuestionnaireChoice(letter="B", statement="Yessir, they do"),
            QuestionnaireChoice(letter="C", statement="Github actions is a version control system"),
            QuestionnaireChoice(letter="D", statement="I don't know these two technologies")
        ),
        answer="B",
        tag="devops"
    ),
    QuestionnaireQuestion(
        statement="Docker was here long before Kubernetes",
        choices= c(
            QuestionnaireChoice(letter="A", statement="This is a big ass lie"),
            QuestionnaireChoice(letter="B", statement="That's a fact"),
            QuestionnaireChoice(letter="C", statement="Both were developed around the same time"),
            QuestionnaireChoice(letter="D", statement="I have no idea what Docker or Kubernetes is")
        ),
        answer="A",
        tag="devops"
    ),
    QuestionnaireQuestion(
        statement="Prometheus is famous for monitoring server applications",
        choices= c(
            QuestionnaireChoice(letter="A", statement="I agree"),
            QuestionnaireChoice(letter="B", statement="Statement is false"),
            QuestionnaireChoice(letter="C", statement="It only collects information"),
            QuestionnaireChoice(letter="D", statement="Didn't catch that")
        ),
        answer="A",
        tag="devops"
    ),
    QuestionnaireQuestion(
        statement="Kali is a Linux distro",
        choices= c(
            QuestionnaireChoice(letter="A", statement="Not!"),
            QuestionnaireChoice(letter="B", statement="It's a flavor of Ubuntu"),
            QuestionnaireChoice(letter="C", statement="It is!"),
            QuestionnaireChoice(
                letter="D", statement="I don't mess with Linux")
        ),
        answer="C",
        tag="devops"
    ),
    QuestionnaireQuestion(
        statement="What does Google Cloud Platform, Azure and Heroku have in common?",
        choices= c(
            QuestionnaireChoice(letter="A", statement="All have IaaS solutions"),
            QuestionnaireChoice(letter="B", statement="They're the top cloud providers"),
            QuestionnaireChoice(letter="C", statement="They all provide PaaS solutions"),
            QuestionnaireChoice(letter="D", statement="They nothing in common")
        ),
        answer="C",
        tag="devops"
    )
)


mobileQuestions<- c(
    QuestionnaireQuestion(
        statement="It's cheaper to host my applications on Apple app store than Google play store",
        choices= c(
            QuestionnaireChoice(letter="A", statement="That's a lie my guy"),
            QuestionnaireChoice(
                letter="B", statement="Sure, Apple offer the cheapest hosting for mobile apps"),
            QuestionnaireChoice(letter="C", statement="They offer almost the same pricing"),
            QuestionnaireChoice(letter="D", statement="I don't host my apps here")
        ),
        answer="A",
        tag="mobile"
    ),
    QuestionnaireQuestion(
        statement="Which one of the following describes what Swift does?",
        choices= c(
            QuestionnaireChoice(letter="A", statement="It's used for developing iOS apps"),
            QuestionnaireChoice(
                letter="B", statement="It can be used for developing Safari web extensions"),
            QuestionnaireChoice(letter="C", statement="Second name of a very famous music artist"),
            QuestionnaireChoice(letter="D", statement="I don't know anything about Swift")
        ),
        answer="A",
        tag="mobile"
    ),
    QuestionnaireQuestion(
        statement="Cross platform development allows me to implement one design for all my apps",
        choices= c(
            QuestionnaireChoice(
                letter="A", statement="You have to implement different designs using native UI SDKs"),
            QuestionnaireChoice(
                letter="B", statement="This is true, the design will be the same for all apps"),
            QuestionnaireChoice(
                letter="C", statement="It's one codebase but designs come out differently"),
            QuestionnaireChoice(letter="D", statement="I have no idea")
        ),
        answer="B",
        tag="mobile"
    ),
    QuestionnaireQuestion(
        statement="Large scale mobile applications can be built without user journeys and design prototypes",
        choices= c(
            QuestionnaireChoice(
                letter="A", statement="User journeys are not required for large projects"),
            QuestionnaireChoice(
                letter="B", statement="That's impossible, user's journeys should be a prerequisite"),
            QuestionnaireChoice(
                letter="C", statement="Design prototypes are created at the same time an MVP is developed"),
            QuestionnaireChoice(
                letter="D", statement="I don't do mobile development")
        ),
        answer="B",
        tag="mobile"
    ),
    QuestionnaireQuestion(
        statement="Which one of the following best describes what Java SDK is for?",
        choices= c(
            QuestionnaireChoice(letter="A", statement="It's for developing backend applications"),
            QuestionnaireChoice(
                letter="B", statement="It's for developing different versions of Java"),
            QuestionnaireChoice(letter="C", statement="It's used for android app development"),
            QuestionnaireChoice(letter="D", statement="I'm not well versed with Java")
        ),
        answer="C",
        tag="mobile"
    )
)
